### PR TITLE
[python] Fix python GUIs and add NederdroneStatus

### PR DESCRIPTION
**Pull request by dewagter**
Hydrogen status and CAN motor status for the Nederdrone.

### CODE REVIEW

**Comment by Fabien-B** [comment]
Could you move the program from control_panel.xml to an independent file in conf/tools ?
The programs in control_panel.xml are only kept for compatibility reasons, and for personal configuration.

**Comment by Fabien-B** [comment]
Also, if I test it:
``` 
File "/home/fabien/paparazzi/sw/ground_segment/python/fuelcell/fuel_cell_viewer.py", line 440
    dx = int(0.43*w)
                   ^
TabError: inconsistent use of tabs and spaces in indentation
```

**Comment by fvantienen** [comment]
This is also missing the generic uart module at the autopilot side.

**Comment by Fabien-B** [comment]
With python3 and wx.__version__ = '4.0.7', I have this error:
```
Traceback (most recent call last):
  File "sw/ground_segment/python/fuelcell/fuel_cell.py", line 26, in OnInit
    self.main = fuel_cell_viewer.FuelCellFrame()
  File "/home/fabien/paparazzi/sw/ground_segment/python/fuelcell/fuel_cell_viewer.py", line 506, in __init__
    self.SetPosition(wx.Point(self.x,self.y), wx.SIZE_USE_EXISTING)
TypeError: Window.SetPosition(): too many arguments
OnInit returned false, exiting...
```

**Comment by dewagter** [comment]
> This is also missing the generic uart module at the autopilot side.

The CAN bus motor status works and the fuel-cell serial is only added when needed.

**Comment by dewagter** [comment]
With python3 I can not test... will have to update my machine I'm afraid.

```
RUN '/home/mavlab/paparazzi/sw/ground_segment/python/fuelcell/fuel_cell.py '
Error: Ivy requires Python 3.8 or later.
```

**Comment by dewagter** [comment]
Is Ubuntu 18.04.4 LTS does not seem to like python3.8
I could finally install it but then pip installing wx does not work.
Does this mean 18.04 is officially not supported anymore? (due to the PPRZLINK Python>v3.8 requirement)?

**Comment by Fabien-B** [comment]
Its a bug, it should works. I do not have a 1804 anymore to test, but how did you install ivy ?

**Comment by dewagter** [comment]
I installed all my paparazzi installations with the 1-line command from the
wiki. Looking at the files this one looks like a feb 2019 installation of
18.04 LTS.

-Christophe

**Comment by gautierhattenberger** [comment]
The ivy python Debian package is still python 2.x, so using ivy-python with python 3 requires to use pip install. For Ubuntu 20.04, the ivy-python package is directly the python 3 version

File: sw/ground_segment/python/fuelcell/fuel_cell_viewer.py [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+        elif int(self.status == 4):
+            return 'Battery only'
+        else:
+            return self.status
+    def get_status_perc(self):
+        return 0.0
+    def get_status_color(self):
+        if (self.status) < 1:
+            return 0.1
+        elif (self.status) == 1:
```
Comment (Fabien-B) [inline sw/ground_segment/python/fuelcell/fuel_cell_viewer.py:294]: self.status seems to be a string, not an int. (python is raising an exception)

File: sw/ground_segment/python/fuelcell/fuel_cell_viewer.py [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+            self.w = int(self.cfg.Read('width'))
+            self.h = int(self.cfg.Read('height'))
+
+        wx.Frame.__init__(self, id=-1, parent=None, name=u'FuelCellFrame',
+                          size=wx.Size(self.w, self.h), title=u'Fuel Cell Monitoring')
+
+        if self.cfg.Exists('left'):
+            self.x = int(self.cfg.Read('left'))
+            self.y = int(self.cfg.Read('top'))
+            self.SetPosition(wx.Point(self.x,self.y), wx.SIZE_USE_EXISTING)
```
Comment (Fabien-B) [inline sw/ground_segment/python/fuelcell/fuel_cell_viewer.py:506]: this method only take one argument, the second one should be removed.

Comment (dewagter) [reply sw/ground_segment/python/fuelcell/fuel_cell_viewer.py]: Removed in all python tools that have this

**Comment by dewagter** [comment]
I installed a fresh Ubuntu 20.04, added python3-wxgtk4.0 and required https://github.com/paparazzi/paparazzi/pull/2729

I fixed all python apps.

File: sw/ground_segment/python/fuelcell/fuel_cell_viewer.py [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+    def get_status(self):
+        if int(self.status == 0):
+            return 'Starting'
+        elif int(self.status == 1):
+            return 'Ready (wait for button press)'
+        elif int(self.status == 2):
+            return 'Running'
+        elif int(self.status == 3):
+            return 'Fault'
+        elif int(self.status == 4):
```
Comment (Fabien-B) [inline sw/ground_segment/python/fuelcell/fuel_cell_viewer.py:285]: Are you sure those  end parenthesis are at the right place?

File: sw/ground_segment/python/fuelcell/fuel_cell_viewer.py [hunk 4]
```diff
@@ -0,0 +1,548 @@
...
+            return 'Battery only'
+        else:
+            return self.status
+    def get_status_perc(self):
+        return 0.0
+    def get_status_color(self):
+        if int(self.status) < 1:
+            return 0.1
+        elif (self.status) == 1:
```
Comment (Fabien-B) [inline sw/ground_segment/python/fuelcell/fuel_cell_viewer.py:294]: also convert to int?

**Review by Fabien-B — APPROVED** [review]

**Pull request merged**